File: Problem2_SumRootToLeaf.java
/************ Time Complexity: O(h) *************/
/************ Space Complexity: O(1) *************/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return iterativeSum(root, 0); //current sum = 0
    }
    
    public int iterativeSum(TreeNode node, int val){
        //base case
        if(node == null)
            return 0;
        
        //leaf node
        if(node.left == null && node.right==null) 
            return val*10 + node.val;
        //not a leaf node
        else
            //call iteravtively to left and right of the node and calculate the sum
            return iterativeSum(node.left, val*10 + node.val)+
                iterativeSum(node.right, val*10 + node.val);
    }
}

File: Problem2_SumRootToLeaf.cs

// Time Complexity :O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int sum = 0;
    public int SumNumbers(TreeNode root) {
        FindTreeSum(root, 0);
        return sum;
    }

    private void FindTreeSum(TreeNode root, int curr) {
        if(root == null) return;

        //Compute current value till leaf node is found.
        curr = curr * 10 + root.val;
        if(root.left == null && root.right == null) {
            //if leaf node is found add the computed value to sum.
            sum += curr;
        }
        //Traverse left
        FindTreeSum(root.left, curr);
        //Traverse right       
        FindTreeSum(root.right, curr);
    }
}